src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs

```diff
@@ -18,6 +18,11 @@ internal sealed class PooledByteBufferWriter : IBufferWriter<byte>, IDisposable
 
         private const int MinimumBufferSize = 256;
 
+        private PooledByteBufferWriter()
+        {
+            _rentedBuffer = null!;
```

Comment:
To avoid having to lie about the nullable annotations and potentially having code elsewhere that erroneously dereferences null, should we use `Array.Empty<byte>()` instead?

Reply:
Actually, given that the class is now effectively bimodal (has a buffer that should be returned vs. is cleared/disposed) and comes with methods that are exclusive to each state, I think it might make sense to explicitly mark the field as nullable and make relevant assertions in each API call.